Basic SvelteKit Examples

CTNicholas opened the pull request:
## Info
I've added two SvelteKit examples to the repo, one for _live cursors_ and one for _live avatars_. Inside each example, under `src/[examplename].svelte`, is a commented file briefly explaining the inner workings. Both of these examples make use of some quick hooks I've built using Svelte stores, with an API that mostly reflects the `liveblocks-react` package. More info can be found inside the packages, within `src/lib-liveblocks`.

### Live cursors
The [SvelteKit live cursors repo](https://github.com/liveblocks/liveblocks/tree/main/examples/sveltekit-live-cursors) contains a replica of the [basic live cursors example](https://liveblocks.io/examples/live-cursors-basic).

### Live avatars
The [SvelteKit live avatars repo](https://github.com/liveblocks/liveblocks/tree/main/examples/sveltekit-live-avatars) contains a replica of the [live avatar stack example](https://liveblocks.io/examples/live-avatar-stack).

GuillaumeSalles commented on examples/sveltekit-live-avatars/README.md:
```diff
@@ -0,0 +1,35 @@
+
+# [Liveblocks](https://liveblocks.io) × [SvelteKit](https://kit.svelte.dev/).
+
+This repo shows how to implement live avatars with Liveblocks and [SvelteKit](https://kit.svelte.dev/). [Demo](https://sveltekit-live-avatars.vercel.app/)
```
I'm wondering if we should link to liveblocks.io avatars example since it's the exact same output (even if they are not built with Svelte) 🤔

GuillaumeSalles commented on examples/sveltekit-live-avatars/README.md:
```diff
@@ -0,0 +1,35 @@
+
+### Run examples on CodeSandbox
+
+- Open this repository on CodeSandbox with this [link](https://codesandbox.io/s/github/liveblocks/liveblocks/tree/main/examples/nextjs-live-avatars)
```
Should link to `sveltekit-live-avatars`

GuillaumeSalles commented on examples/sveltekit-live-avatars/package.json:
```diff
@@ -0,0 +1,31 @@
+  "devDependencies": {
+    "@sveltejs/adapter-auto": "next",
+    "@sveltejs/kit": "next",
+    "@types/cookie": "^0.4.1",
```
Is it used?

GuillaumeSalles commented on examples/sveltekit-live-avatars/package.json:
```diff
@@ -0,0 +1,31 @@
+    "svelte": "^3.46.0",
+    "svelte-check": "^2.2.6",
+    "svelte-preprocess": "^4.10.1",
+    "tailwindcss": "^3.0.23",
```
Are we using tailwindcss only for the example info box? I'm wondering if we should build it with in css only to reduce the noise for developers that will clone the example.

GuillaumeSalles commented on examples/sveltekit-live-avatars/src/lib-liveblocks/useOthers.ts:
```diff
@@ -0,0 +1,31 @@
+ * Works similarly to `liveblocks-react` useOthers
+ * https://liveblocks.io/docs/api-reference/liveblocks-react#useOthers
+ *
+ * This uses a vue ref so make sure to use .value
```
I think it's copy paste from your vue example :D

GuillaumeSalles reviewed (commented):
That's awesome @CTNicholas! So great that you managed to integrate with Svelte so fast! I love that you extracted the Liveblocks logic as Svelte store! It's good first step toward an official @liveblocks/svelte package!

CTNicholas commented:
Hi! I've fixed everything above, and added `.prettierrc` config files to all packages, for automatic formatting. There's also a couple more Svelte hooks now.

GuillaumeSalles reviewed (approved):
Perfect! Thank you for taking the time to introduce prettier config everywhere. 🙇‍♂️
